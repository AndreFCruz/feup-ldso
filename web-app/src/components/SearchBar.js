import React, { Component } from 'react';
import { Input, Row, Col } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';

import '../styles/SideBar.css';

class SearchBar extends Component {
    constructor(props) {
        super(props);

        this.state = {
            firstRequest: false
        };
    }

    render() {
        return (
            <Row>
                <Col className="searchBar" >
                    <FontAwesomeIcon className="search-icon" icon="search" />
                    <Input placeholder="search" />
                </Col>
            </Row>
        );
    }
}

SearchBar.propTypes = {
};

export default SearchBar;
