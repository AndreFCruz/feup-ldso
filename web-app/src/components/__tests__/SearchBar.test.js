import React from 'react';
import SearchBar from '../SearchBar';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


it('renders sidebar correctly', () => {
    const wrapper = shallow(<SearchBar />);

    expect(wrapper).toMatchSnapshot();
});