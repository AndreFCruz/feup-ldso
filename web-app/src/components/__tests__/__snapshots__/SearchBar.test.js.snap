// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders without crashing 1`] = `
<div>
  <div
    className="intro"
  >
    <img
      alt="application logo"
      src="test-file-stub"
      width="140"
    />
  </div>
  <div>
    <div
      className="navigator"
    >
      <h6
        className="fadeup"
      >
        <a
          aria-current={null}
          href="/faculties/fadeup"
          onClick={[Function]}
        >
          FADEUP
        </a>
      </h6>
      <h6
        className="faup"
      >
        <a
          aria-current={null}
          href="/faculties/faup"
          onClick={[Function]}
        >
          FAUP
        </a>
      </h6>
      <h6
        className="fbaup"
      >
        <a
          aria-current={null}
          href="/faculties/fbaup"
          onClick={[Function]}
        >
          FBAUP
        </a>
      </h6>
      <h6
        className="fcnaup"
      >
        <a
          aria-current={null}
          href="/faculties/fcnaup"
          onClick={[Function]}
        >
          FCNAUP
        </a>
      </h6>
      <h6
        className="fcup"
      >
        <a
          aria-current={null}
          href="/faculties/fcup"
          onClick={[Function]}
        >
          FCUP
        </a>
      </h6>
      <h6
        className="fdup"
      >
        <a
          aria-current={null}
          href="/faculties/fdup"
          onClick={[Function]}
        >
          FDUP
        </a>
      </h6>
      <h6
        className="fep"
      >
        <a
          aria-current={null}
          href="/faculties/fep"
          onClick={[Function]}
        >
          FEP
        </a>
      </h6>
      <h6
        className="feup"
      >
        <a
          aria-current={null}
          href="/faculties/feup"
          onClick={[Function]}
        >
          FEUP
        </a>
      </h6>
      <h6
        className="ffup"
      >
        <a
          aria-current={null}
          href="/faculties/ffup"
          onClick={[Function]}
        >
          FFUP
        </a>
      </h6>
      <h6
        className="flup"
      >
        <a
          aria-current={null}
          href="/faculties/flup"
          onClick={[Function]}
        >
          FLUP
        </a>
      </h6>
      <h6
        className="fmdup"
      >
        <a
          aria-current={null}
          href="/faculties/fmdup"
          onClick={[Function]}
        >
          FMDUP
        </a>
      </h6>
      <h6
        className="fmup"
      >
        <a
          aria-current={null}
          href="/faculties/fmup"
          onClick={[Function]}
        >
          FMUP
        </a>
      </h6>
      <h6
        className="fpceup"
      >
        <a
          aria-current={null}
          href="/faculties/fpceup"
          onClick={[Function]}
        >
          FPCEUP
        </a>
      </h6>
      <h6
        className="icbas"
      >
        <a
          aria-current={null}
          href="/faculties/icbas"
          onClick={[Function]}
        >
          ICBAS
        </a>
      </h6>
    </div>
    <div
      className="container null"
    >
      <div>
        <h1>
          Redirect to App login or another admin page
        </h1>
      </div>
    </div>
  </div>
  <div
    className="notification-container notification-container-empty"
  >
    <span />
  </div>
</div>
`;

exports[`renders sidebar correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Col
        className="searchBar"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <FontAwesomeIcon
          border={false}
          className="search-icon"
          fixedWidth={false}
          flip={null}
          icon="search"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
        <DebounceInput
          className="form-control"
          debounceTimeout={1000}
          element="input"
          forceNotifyByEnter={true}
          forceNotifyOnBlur={true}
          minLength={2}
          onChange={[Function]}
          placeholder="search"
          type="text"
        />
      </Col>,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <FontAwesomeIcon
            border={false}
            className="search-icon"
            fixedWidth={false}
            flip={null}
            icon="search"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />,
          <DebounceInput
            className="form-control"
            debounceTimeout={1000}
            element="input"
            forceNotifyByEnter={true}
            forceNotifyOnBlur={true}
            minLength={2}
            onChange={[Function]}
            placeholder="search"
            type="text"
          />,
        ],
        "className": "searchBar",
        "tag": "div",
        "widths": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "border": false,
            "className": "search-icon",
            "fixedWidth": false,
            "flip": null,
            "icon": "search",
            "inverse": false,
            "listItem": false,
            "mask": null,
            "pull": null,
            "pulse": false,
            "rotation": null,
            "size": null,
            "spin": false,
            "symbol": false,
            "title": "",
            "transform": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "form-control",
            "debounceTimeout": 1000,
            "element": "input",
            "forceNotifyByEnter": true,
            "forceNotifyOnBlur": true,
            "inputRef": undefined,
            "minLength": 2,
            "onBlur": undefined,
            "onChange": [Function],
            "onKeyDown": undefined,
            "placeholder": "search",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Col
          className="searchBar"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <FontAwesomeIcon
            border={false}
            className="search-icon"
            fixedWidth={false}
            flip={null}
            icon="search"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          <DebounceInput
            className="form-control"
            debounceTimeout={1000}
            element="input"
            forceNotifyByEnter={true}
            forceNotifyOnBlur={true}
            minLength={2}
            onChange={[Function]}
            placeholder="search"
            type="text"
          />
        </Col>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <FontAwesomeIcon
              border={false}
              className="search-icon"
              fixedWidth={false}
              flip={null}
              icon="search"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />,
            <DebounceInput
              className="form-control"
              debounceTimeout={1000}
              element="input"
              forceNotifyByEnter={true}
              forceNotifyOnBlur={true}
              minLength={2}
              onChange={[Function]}
              placeholder="search"
              type="text"
            />,
          ],
          "className": "searchBar",
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "search-icon",
              "fixedWidth": false,
              "flip": null,
              "icon": "search",
              "inverse": false,
              "listItem": false,
              "mask": null,
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "title": "",
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "form-control",
              "debounceTimeout": 1000,
              "element": "input",
              "forceNotifyByEnter": true,
              "forceNotifyOnBlur": true,
              "inputRef": undefined,
              "minLength": 2,
              "onBlur": undefined,
              "onChange": [Function],
              "onKeyDown": undefined,
              "placeholder": "search",
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
